pkg/converter/cloudevents/cloudevents.go

```diff
@@ -73,11 +81,71 @@ func (ce *CloudEvent) Convert(data []byte) ([]byte, error) {
 		Time:        time.Now().Format(time.RFC3339),
 		Source:      ce.Source,
 		Specversion: "1.0",
+		Contenttype: contentType,
 		Data:        body,
 	}
 	return json.Marshal(b)
 }
 
+func (ce *CloudEvent) Request(request []byte, headers http.Header) ([]byte, map[string]string, error) {
+	var context map[string]string
+	var body []byte
+	var err error
+
+	switch headers.Get("Content-Type") {
+	case "application/cloudevents+json":
+		if body, context, err = parseStructuredCE(request); err != nil {
+			return nil, nil, fmt.Errorf("structured CloudEvent parse error: %w", err)
+		}
+	case "application/json":
+		body = request
+		context = parseBinaryCE(headers)
+	default:
+		return request, nil, nil
+	}
+
+	ceContext, err := json.Marshal(context)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot encode request context: %w", err)
+	}
+
+	runtimeContext := map[string]string{
+		"Lambda-Runtime-Cloudevents-Context": string(ceContext),
+	}
+
+	return body, runtimeContext, nil
+}
+
+func parseStructuredCE(body []byte) ([]byte, map[string]string, error) {
+	var event map[string]interface{}
+	if err := json.Unmarshal(body, &event); err != nil {
+		return nil, nil, fmt.Errorf("cannot unmarshal body: %w", err)
+	}
+
+	data, err := json.Marshal(event["data"])
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot marshal body: %w", err)
+	}
+
+	delete(event, "data")
+	headers := make(map[string]string, len(event))
+	for k, v := range event {
+		headers[k] = fmt.Sprintf("%v", v)
+	}
+
+	return data, headers, nil
+}
+
+func parseBinaryCE(headers http.Header) map[string]string {
+	h := make(map[string]string)
+	for k, v := range headers {
+		if strings.HasPrefix(k, "Ce-") {
+			h[k[3:]] = strings.Join(v, " ")
```

Comment:
RFC 2616 says values should be separated using commas, not spaces.

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list.

There is no standard for multiple values in a single CE attribute, but since values are received over HTTP, maybe we should also use commas?